### ISSUE TITLE
archived group channel exception

### ISSUE THREAD

**Issue by amaloz** [matterhorn-chat/matterhorn#351]
Archived group channels produce "unexpected error" exceptions and remain highlighted.  To reproduce:

1. Join a group channel.
2. Have one of the members archive it.
3. You should now have the following error: `An unexpected error has occurred! The exception encountered was: MattermostError {mattermostErrorId = "api.context.permissions.app_error", mattermostErrorMessage = "You do not have the appropriate permissions", mattermostErrorRequestId = "5y8iesckoib1tpejrma4kzmiao", mattermostErrorStatusCode = 403, mattermostErrorIsOAuth = False}`

**Comment by jtdaugherty** [comment]
After (2), did anything happen to trigger (3)? As in, did the user who saw that error try to post a message to the channel?

**Comment by amaloz** [comment]
I triggered it by focusing on that channel.  Every time I refocused on the channel, that exception popped up.

**Comment by jtdaugherty** [comment]
Okay, so:

1) Users A and B join a group channel.
2) B archives the channel.
3) A returns to the channel and sees the aforementioned exception.

**Comment by jtdaugherty** [comment]
For step 2, is that done in the web client, or Matterhorn? In either case, what action was taken (or what command was executed) to do the archiving?

**Comment by amaloz** [comment]
Yes, except this happened in a group channel (not sure if that'll matter for reproducibility).  So, I'd say the first step is:

1. Users A, B, and C join a group channel.

The archiving was done in matterhorn using `/delete`

**Comment by jtdaugherty** [comment]
Okay, that helps. Thanks!

**Comment by jtdaugherty** [comment]
I was unable to reproduce this with the following steps:
* `/group-msg a b`
* Go to new group channel with users `a` and `b`
* `/delete-channel`, confirm
* Now users `a` and `b` can still post to the channel

Is that the workflow you had in mind?

**Comment by jtdaugherty** [comment]
@amaloz do you have any thoughts on whether the above workflow matches what you saw?

**Comment by amaloz** [comment]
That's basically it, although I believe a user that did not create the group message was the one that deleted it.  Maybe that'd trigger the bug?

**Comment by jtdaugherty** [comment]
The thing is, group channels *cannot* be deleted. Internally, Matterhorn knows this, and `/delete-channel` just *leaves* the channel. (It will reappear if anyone in the channel sends a message to it.)

**Comment by amaloz** [comment]
In the case above I don't believe I was able to post anything. It was as if the channel did not exist (maybe hence the `you do not have appropriate permissions` error)?

**Comment by jtdaugherty** [comment]
Okay. That's why I'm trying to be absolutely sure that this was a *group channel* rather than a normal or private one. Since group channels cannot be deleted, it made me wonder whether this was really about one of those, or about a normal channel.

**Comment by jtdaugherty** [comment]
Here's another question: do you recall whether the person who *archived* the channel was using the web client?

**Comment by amaloz** [comment]
Hmm, I was not aware of the distinction. Since I didn't create the channel in the first place, I'm not sure what type of channel is was. And I'm not sure how it was deleted, sorry.

**Comment by jtdaugherty** [comment]
You'll know it's a "group" channel if the name is just a bunch of usernames of the members.

**Comment by amaloz** [comment]
Ah okay, so it was a group channel.

**Comment by jtdaugherty** [comment]
I'm wondering about whether the "archiver" used the web client because it's possible the web client did something different than what Matterhorn does, due to us not knowing the right way to handle an archival request for a group channel.
